core/src/main/java/org/jenkins/ui/icon/IconSet.java

```diff
@@ -73,34 +75,40 @@ private static String prependTitleIfRequired(String icon, String title) {
         return icon;
     }
 
-    public static String getIonicon(String name, String title) {
-        if (IONICONS.containsKey(name)) {
-            String icon = IONICONS.get(name);
-            return prependTitleIfRequired(icon, title);
+    // for Jelly
+    @Restricted(NoExternalUse.class)
+    public static String getSymbol(String name, String title, String classes) {
+        if (SYMBOLS.containsKey(name)) {
+            String symbol = SYMBOLS.get(name);
+            symbol = symbol.replaceAll("(class=\")[^&]*?(\")", "$1$2");
+            symbol = symbol.replaceAll("<svg", "<svg class=\"" + classes + "\"");
+            return prependTitleIfRequired(symbol, title);
         }
 
-        // Load icon if it exists
-        InputStream inputStream = IconSet.class.getResourceAsStream("/images/ionicons/" + name + ".svg");
-        String ionicon = null;
+        // Load symbol if it exists
+        InputStream inputStream = IconSet.class.getResourceAsStream("/images/symbols/" + name + ".svg");
+        String symbol = null;
 
         try {
             if (inputStream != null) {
-                ionicon = IOUtils.toString(inputStream, StandardCharsets.UTF_8);
+                symbol = IOUtils.toString(inputStream, StandardCharsets.UTF_8);
             }
         } catch (IOException e) {
             // ignored
         }
-        if (ionicon == null) {
-            ionicon = PLACEHOLDER_SVG;
+        if (symbol == null) {
+            symbol = PLACEHOLDER_SVG;
         }
 
-        ionicon = ionicon.replaceAll("(<title>)[^&]*(</title>)", "$1$2");
-        ionicon = ionicon.replaceAll("<svg", "<svg aria-hidden=\"true\"");
-        ionicon = ionicon.replace("stroke:#000", "stroke:currentColor");
+        symbol = symbol.replaceAll("(<title>)[^&]*(</title>)", "$1$2");
+        symbol = symbol.replaceAll("(class=\")[^&]*?(\")", "$1$2");
+        symbol = symbol.replaceAll("<svg", "<svg aria-hidden=\"true\"");
+        symbol = symbol.replaceAll("<svg", "<svg class=\"" + classes + "\"");
+        symbol = symbol.replace("stroke:#000", "stroke:currentColor");
```

Comment:
Seems that this replacement does not work for non ionicon icons? I tried with `stroke:currentColor` in my SVG sidebar links but they still appear in black. What is required to get the correct color automatically? It would be helpful if this could work out-of-the box here as well:

![Bildschirmfoto 2022-01-30 um 13 01 56](https://user-images.githubusercontent.com/503338/151698872-855eea07-4f55-4ac9-806e-bf842cc3608b.png)

Reply:
Sadly there isn't any support for plugins yet. For adding Symbols to Jenkins, it should just work by adding them to the `symbols` folder - some SVGs may need a little tweaking (eg manually updating the `fill` attribute if it hasn't already been set) but for the most part they should just work automatically.